script/test_in_openshift/lib.sh

```diff
@@ -0,0 +1,231 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+function convert::print_msg () {
+    echo ""
+    tput setaf 4
+    tput bold
+    echo -e "$@"
+    tput sgr0
+    echo ""
+}
+
+function install_oc_client () {
+    # Valid only for Travis
+    convert::print_msg "Installing oc client binary ..."
+    sudo sed -i 's:DOCKER_OPTS=":DOCKER_OPTS="--insecure-registry 172.30.0.0/16 :g' /etc/default/docker
+    sudo mv /bin/findmnt /bin/findmnt.backup
+    sudo /etc/init.d/docker restart
+    # FIXME
+    wget https://github.com/openshift/origin/releases/download/v1.4.1/openshift-origin-client-tools-v1.4.1-3f9807a-linux-64bit.tar.gz -O /tmp/oc.tar.gz 2> /dev/null > /dev/null
+    mkdir /tmp/ocdir && cd /tmp/ocdir && tar -xvvf /tmp/oc.tar.gz > /dev/null
+    sudo mv /tmp/ocdir/*/oc /usr/bin/
+}
+
+
+function convert::oc_cluster_up () {
+
+    oc cluster up; exit_status=$?
+
+    if [ $exit_status -ne 0 ]; then
+	FAIL_MSGS=$FAIL_MSGS"exit status: $exit_status\n";
+	convert::print_fail "oc cluster up failed.\n"
+	exit $exit_status
+    fi
+
+    convert::run_cmd "oc login -u system:admin"
+}
+
+function convert::oc_cluster_down () {
+
+    convert::run_cmd "oc cluster down"
+    exit_status=$?
+
+    if [ $exit_status -ne 0 ]; then
+	FAIL_MSGS=$FAIL_MSGS"exit status: $exit_status\n"
+	exit $exit_status
+    fi
+
+}
+
+function convert::kompose_up () {
+    dc_file=$1
+    kompose_cli='kompose --emptyvols --provider=openshift -f $dc_file up'
+    convert::run_cmd "${kompose_cli}"
+    exit_status=$?
+
+    if [ $exit_status -ne 0 ]; then
+	FAIL_MSGS=$FAIL_MSGS"kompose up failed with exit status: $exit_status\n"
+	exit $exit_status
+    fi
+}
+
+function convert::kompose_down () {
+    dc_file=$1
+    kompose_cli='kompose --emptyvols --provider=openshift -f $dc_file down'
+    convert::run_cmd $kompose_cli
+    exit_status=$?
+
+    if [ $exit_status -ne 0 ]; then
+	FAIL_MSGS=$FAIL_MSGS"kompose down failed with exit status: $exit_status\n";
+	exit $exit_status;
+    fi
+}
+
+
+function convert::kompose_up_check () {
+    # Usage: -p for pod name, -r replica count
+    local retry_up=0
+   
+    while getopts ":p:r:" opt; do
+	case $opt in
+	    p ) pod=$OPTARG;;
+	    r ) replica=$OPTARG;;
+	esac
+    done
+
+    if [ -z $replica ]; then
+       replica_1=1
+       replica_2=1
+    else
+       replica_1=$replica
+       replica_2=$replica
+    fi
+    
+    pod_1=$( echo $pod | awk '{ print $1 }')
```

Comment:
I don't like that :-(
If you are not going to fix that, please write some big warning that this is the limitation and also TODO that this needs to be fixed. It will cause a lot of confusion for anyone writing tests.